CDTDatastore/touchdb/TD_Database+Insertion.m

```diff
@@ -391,7 +396,7 @@ - (TD_Revision*)putRevision:(TD_Revision*)rev
         } else {
             // Inserting first revision, with no docID given (POST): generate a unique docID:
             docID = [[self class] generateDocumentID];
-            docNumericID = [self insertDocumentID:docID inDatabase:db];
+            docNumericID = [self insertDocumentID:docID inDatabase:db error:NULL];
```

Comment:
Wouldn't we be better making use of the error to differentiate between db full errors and general db errors as done elsewhere?